### ISSUE TITLE
Feature request for Case-Insensitive Filtering in completer extension

### ISSUE THREAD

**Issue by geonilpark** [jupyterlab/jupyterlab#4567]
Hi,

I noticed a filtering feature is implemented with Case-Sensitive in completer extension.

For example, without using filtering I have following list:
![image](https://user-images.githubusercontent.com/14316384/40025262-e011b758-5785-11e8-8163-42d39f9f97e2.png)

But when I typed "t", I've got following...
![image](https://user-images.githubusercontent.com/14316384/40025308-0a3a0f8a-5786-11e8-813a-f8ca2498c06a.png)

I would like to make this filtering with Case-Insensitive and I am wondering if this feature can be implemented?

Thank you,
--Geonil

**Comment by adunlea** [comment]
Are there any plans to implement this feature?

Also, I imagine we would also want Case-Insensitive filtering to be optional. For languages like Python Case-Sensitive filtering might be preferred for some users because case matters in that language.

Thanks,
Allen

**Comment by GF-Huang** [comment]
Yes, case sensitivity is important in Python, but case sensitivity in autocomplete is anti-human. My pinkie hurts!
